charts/karpenter/templates/provisioner.yaml

```
@@ -0,0 +1,15 @@
+{{ range .Values.provisioners }}
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: {{ .name }}
```

Thoughts on aligning w/ the k8s API
```suggestion
  name: {{ metadata.name }}
```

---

Actually, the `{{ .name }}` is the `name` key for the yaml object feeded to helm, as shown in the values.yaml sample : 

```
provisioners: {}
  # - name: sample-1
  #   spec:
 ```

---

Yeah. I'm recommending
```
    - metadata:
        name: sample-1
      spec: ...
```
This allows you to serialize conformant kubernetes objects into the chart.

---

Ok got it, updated code and values